addons/mail/static/src/core/web/thread_model_patch.js

```diff
@@ -11,23 +11,9 @@ patch(Thread.prototype, {
     setup() {
         super.setup();
         this.recipients = Record.many("Follower");
+        this.activities = Record.many("Activity");
     },
     get recipientsFullyLoaded() {
         return this.recipientsCount === this.recipients.length;
     },
-    /**
-     * @returns {import("models").Activity[]}
-     */
-    get activities() {
-        return Object.values(this._store.Activity.records)
-            .filter((activity) => {
-                return activity.res_model === this.model && activity.res_id === this.id;
-            })
-            .sort(function (a, b) {
-                if (a.date_deadline === b.date_deadline) {
-                    return a.id - b.id;
-                }
-                return a.date_deadline < b.date_deadline ? -1 : 1;
-            });
```

Comment:
Shouldn't the field be sorted then?

Reply:
I don't think we can trust that, because activities can be received on the fly, or the date can be changed, and then the order will no longer be correct, as they are store in insertion order in the relation if the field doesn't define the sort.